src/inpod/statemanager.rs

```diff
@@ -231,6 +231,8 @@ impl WorkloadProxyManagerState {
             .await
         {
             Ok(()) => {
+                // If the workload is already pending, make sure we drop it, so we don't retry.
+                self.pending_workloads.remove(workload_uid);
```

Comment:
I'm beginning to think we should probably key all of the workload stores with UID + NetNSID to avoid future goofiness like this.

Reply:
one nice part about UID is it means we will never have multiple Proxy running for a given pod. I would worry if we have netns ID as a key we could end up with some leaky Proxy instances somehow maybe?

Might be possible, but needs care